INPUTCONFIG.md

```diff
@@ -50,50 +46,63 @@ This can be used to add error information to the input state, such as setting er
-
-#### Arguments
-1. `value` *(Any)* The value of the input being validated
-2. `inputsState` *(Object)* The current state of your inputs
-3. `state` *(Object)* The state of the entire redux store
-4. `dispatch` *(Function)* dispatch function from the redux store, for dispatching side-effect actions
-
-#### Returns
-- *(Promise)*: Resolves if valid, rejects if invalid.
-    - `resolve` *(Function)*
-    - `reject` *(Function)*
-        - `newInputState` *(Object)* Optionally return the new input state object to set the input to. This is useful for adding error information to the input state.
-
-#### Example
-
-    inputConfig: {
         zipCode: {
-            asyncValidator: (value) => new Promise((resolve, reject) => {
-                asyncValidateZIPCode((error) => {
-                    if (error) {
-                        if (error === 'Unsupported') {
-                            reject({
-                                error: value,
-                                errorType: 'Unsupported' // Extra information stored on input state to be used in view
-                            });
+            validator: (value) => {
+                // Client-side validation
+                if (!value || value.length < 5 || value.length > 5) {
+                    return false;
+                }
+                
+                // Async validation
+                return new Promise((resolve, reject) => {
+                    asyncValidateZIPCode((error) => {
+                        if (error) {
+                            if (error === 'Unsupported') {
+                                reject({
+                                    error: value,
+                                    errorType: 'Unsupported' // Extra information stored on input state to be used in view
+                                });
+                            } else {
+                                reject(); // Generic error
+                            }
                         } else {
-                            reject();
+                            resolve(); // Validation passed
                         }
-                    } else {
-                        resolve();
-                    }
-                })
-            })
+                    })
+                });
+            }
         }
     }
 
+### `onChange(inputState, inputs, state, dispatch)`
+ 
+Function for firing actions after an input is validate and changed.
+ 
+#### Passed arguments
+1. `inputState` *(Object)* The new state of the input being changed. Contains properties such as `value`, `error`, `validating`.
+2. `inputs` *(Object)* The new state of your inputs
+3. `state` *(Object)* The new state of the entire redux store
+4. `dispatch` *(Function)* dispatch function from the redux store, for dispatching side-effect actions
+
+### `props` *(Object)* Props to pass down to every component associated with this input 
+
 ## Form Config
```

Comment:
The reducer needs to know which actions to listen to, and reducers don't know where they are mounted in the overall state.